### ISSUE TITLE
URL for Gradle sources is hard-coded into SourceDistributionResolver

### ISSUE THREAD

**Issue by a--v--k** [gradle/gradle#18249]
URL for Gradle sources is hard-coded into SourceDistributionResolver.
So source downloading failed in case of Service.gradle.org is blocked by corporate firewall or tracker software installed on local host.

### Expected Behavior

User must have ability to set custom URL (server name and path for artifact) 
to make Gradle download sources from intranet repositories (Nexus, Artifactory, etc)

or

User must have ability to make Gradle use URL from wrapper.properties in project.

### Current Behavior
Building Kotlin script model, Gradle tries to download sources on each module in project.
If each attempt is failed, it spend a lot of time and make IDEA very slow .

### Steps to Reproduce 
IMHO, it should be reproduced with blocking resolving Service.gradle.org on develepment machine.

### Your Environment
Gradle 6.7.1 - 7.2
Windows 10 developer machines
Project with ~60 modules and ~ 10 internal plugins
Jetbrains IDEA  2020.2

**Comment by jbartok** [comment]
This `SourceDistributionResolver` doesn't seem to be a Gradle class. Are you sure this is not an IntelliJ problem? If so, pls. report it there.

**Comment by kelemen** [comment]
Copying this from my report (#20869):

A workaround for this is to specify the "-all.zip" distribution instead of "-bin.zip" (since the code actually checks for the presence of the *src* directory, before attempting to download the sources).

Also - not replacing the ability to specify a custom sources URL -, a convenient default behavior would be to deduce the sources URL from the distribution URL. Even though it is not technically 100% backward compatible, it should be in realistic use-cases.

**Comment by xenoterracide** [comment]
why isn't this fixed yet? I actually had no luck using -all with intellij

**Comment by kelemen** [comment]
@xenoterracide One thing you have to consider is that after you change it to "-all", you have to kill all daemons, because otherwise Gradle only checks the version (which is the same for all and bin), and will just reuse the "-bin" version.

**Comment by xenoterracide** [comment]
either way you may not want all, all the time, because CI doesn't need things like source and javadoc

**Comment by EarthCitizen** [comment]
I agree with @kelemen that it could, and probably should, drive the src download URL from the distribution URL.

**Comment by EarthCitizen** [comment]
We have made a raw proxy repository for the distribution downloads, but of course this does not work for the src download because it is hard-coded. Perhaps just add another property to `gradle-wrapper.properties`?

It seems like a really bad idea to keep the src download URL unmodifiable.

**Comment by sk23628** [comment]
Any update on this issue? I am facing same problem
Should I use -all instead of -bin?

**Comment by EarthCitizen** [comment]
@sk23628 I think `-all` is the only solution for now.

**Comment by mljohns89** [comment]
Just ran into this issue too LoL

**Comment by mljohns89** [comment]
Nice to see a year later I had the exact same problem, forgot, but thankfully I gave myself the answer already

**Comment by ljacomet** [comment]
Sorry for the late reply!

This issue will be closed as a duplicate of

* #27863

Please add your use case and 👍 to that issue.

If you think our analysis is wrong, please provide us with more detailed information explaining why.

***

Picking the other issue as the one to keep open but will update it with a comment to the workaround shared here.
